packages/create-react-app/index.js

```diff
@@ -172,32 +176,74 @@ function run(root, appName, version, verbose, originalDirectory, template) {
 
   var allDependencies = ['react', 'react-dom', packageToInstall];
 
+  var installDependencies = function(yarnLockExists) {
+    install(allDependencies, verbose, yarnLockExists, function(code, command, args) {
+      if (code !== 0) {
+        console.error(chalk.cyan(command + ' ' + args.join(' ')) + ' failed');
+        process.exit(1);
+      }
+
+      checkNodeVersion(packageName);
+
+      // Since react-scripts has been installed with --save
+      // We need to move it into devDependencies and rewrite package.json
+      if (!yarnLockExists) {
+        moveReactScriptsToDev(packageName);
+      }
+
+      var scriptsPath = path.resolve(
+        process.cwd(),
+        'node_modules',
+        packageName,
+        'scripts',
+        'init.js'
+      );
+      var init = require(scriptsPath);
+      init(root, appName, verbose, originalDirectory, template);
+    });
+  }
+
   console.log('Installing packages. This might take a couple minutes.');
   console.log('Installing ' + chalk.cyan('react, react-dom, ' + packageName) + '...');
   console.log();
 
-  install(allDependencies, verbose, function(code, command, args) {
-    if (code !== 0) {
-      console.error(chalk.cyan(command + ' ' + args.join(' ')) + ' failed');
-      process.exit(1);
-    }
-
-    checkNodeVersion(packageName);
-
-    // Since react-scripts has been installed with --save
-    // We need to move it into devDependencies and rewrite package.json
-    moveReactScriptsToDev(packageName);
-
-    var scriptsPath = path.resolve(
-      process.cwd(),
-      'node_modules',
-      packageName,
-      'scripts',
-      'init.js'
-    );
-    var init = require(scriptsPath);
-    init(root, appName, verbose, originalDirectory, template);
-  });
+  // Try to download the yarn.lock file.
+  if (shouldUseYarn()) {
+    var releaseRequest = request({
+      uri: 'https://api.github.com/repos/facebookincubator/create-react-app/releases/latest',
```

Comment:
We wouldn't be able to publish the updated `yarn.lock` (and `package.json`) with `react-scripts` though, since we'd want the `yarn.lock` to reflect the latest `react-scripts` (we'd only be able to generate it properly after `react-scripts` was published.. And we wouldn't publish it within the CLI since you'd need to have the latest CLI to use the latest `react-scripts` when bootstrapping.

It is a nice to have - if it starts to get too complex / too difficult to maintain, I don't want to force it. Upfront, I thought this would be much simpler than it might actually be.